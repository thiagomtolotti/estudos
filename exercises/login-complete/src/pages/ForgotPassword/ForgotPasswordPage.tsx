"use client";

import { useState } from "react";
import styles from "./ForgotPassword.module.css";
import ForgotPasswordForm from "./ForgotPasswordForm";
import ForgotPasswordConfirm from "./ForgotPasswodConfirm";
import { useForgotPassword } from "@/hooks/useForgotPassword";

export function ForgotPasswordPage() {
	const [emailSent, setEmailSent] = useState("");

	const submit = useForgotPassword();

	function sendEmail(email: string) {
		//Envia o email
		submit(email);

		setEmailSent(email);
	}

	return (
		<div className={styles.container}>
			<h1>Esqueci a senha</h1>

			{!emailSent ? (
				<ForgotPasswordForm submit={sendEmail} />
			) : (
				<ForgotPasswordConfirm email={emailSent} />
			)}
		</div>
	);
}
