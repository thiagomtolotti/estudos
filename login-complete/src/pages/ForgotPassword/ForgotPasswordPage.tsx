"use client";

import { useState } from "react";
import styles from "./ForgotPassword.module.css";
import ForgotPasswordForm from "./ForgotPasswordForm";
import ForgotPasswordConfirm from "./ForgotPasswodConfirm";

export function ForgotPasswordPage() {
	const [emailSent, setEmailSent] = useState<string>("");

	function sendEmail(email: string) {
		//Envia o email

		setEmailSent(email);
	}

	return (
		<div className={styles.container}>
			<h1>Esqueci a senha</h1>

			{!emailSent ? (
				<ForgotPasswordForm onSubmit={sendEmail} />
			) : (
				<ForgotPasswordConfirm email={emailSent} />
			)}
		</div>
	);
}
