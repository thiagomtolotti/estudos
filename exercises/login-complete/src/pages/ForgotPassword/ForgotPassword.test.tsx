import {
	RenderResult,
	fireEvent,
	render,
	screen,
} from "@testing-library/react";
import { ForgotPasswordPage } from "./ForgotPasswordPage";

describe("ForgotPasswordForm", () => {
	let emailInput: HTMLInputElement, sendButton: HTMLButtonElement;
	let container: RenderResult;

	beforeEach(() => {
		container = render(<ForgotPasswordPage />);

		emailInput = screen.getByPlaceholderText("email");
		sendButton = screen.getByRole("button");
	});

	it("Should render correctly", () => {
		expect(container.container).toMatchSnapshot();
	});

	it("Should change the stage when the email is sent", () => {
		const emailSent = "teste@teste.com";

		fireEvent.input(emailInput, { target: { value: emailSent } });
		fireEvent.click(sendButton);

		expect(emailInput).not.toBeInTheDocument();

		const confirmText = screen.getByTestId("confirm-email-sent");

		expect(confirmText).toBeInTheDocument();
		expect(confirmText).toHaveTextContent(emailSent);
	});
});
