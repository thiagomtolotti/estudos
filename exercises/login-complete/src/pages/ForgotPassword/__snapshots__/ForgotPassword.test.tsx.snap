// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordForm Should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Esqueci a senha
    </h1>
    <form>
      <input
        placeholder="email"
        type="email"
      />
      <button>
        Enviar
      </button>
    </form>
  </div>
</div>
`;
